lib/publiccloud/gce.pm

```diff
@@ -170,14 +170,17 @@ sub cleanup {
 
     my $region = $self->{provider_client}->{region};
     my $project = $self->{provider_client}->{project_id};
-    my $instance_id = $self->get_terraform_output(".vm_name.value[0]");
+    my $query = check_var('PUBLIC_CLOUD_SLES4SAP', 1) ? ".hana_name.value[0]" : ".vm_name.value[0]";
+    my $instance_id = $self->get_terraform_output($query);
     # gce provides full serial log, so extended timeout
     if (!check_var('PUBLIC_CLOUD_SLES4SAP', 1) && defined($instance_id)) {
-        if ($instance_id =~ /$self->{resource_name}/) {
-            script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --zone=$region --port=1 > instance_serial.txt", timeout => 180);
+        if ($instance_id ne "") {
```

Comment:
As `$instance_id` is defined we can just use:

```suggestion
        if ($instance_id) {
```

Reply:
Leftover: In a previous version I used 3-ways output, then uniformed in 2. I shall also remove `defined` above.